src/Element/ContactTabContent.php

```diff
@@ -67,9 +63,8 @@ public function getInfo() {
    *   The passed-in element containing the renderable regions in '#content'.
    */
   public static function preRenderTabContent(array $element) {
-    $tab_manager = static::getTabManager();
-    // Check this tab is valid for the contact.
-    if ($element['#tab'] && $tab_manager->verifyTab($element['#tab'], $element['#user'])) {
+    $blocks = $element['#blocks'];
+    if (!empty($blocks)) {
       $layout = $element['#tab']->get('layout') ?: 'contacts_tab_content.stacked';
       $layout_manager = static::getLayoutManager();
       $layoutInstance = $layout_manager->createInstance($layout, []);
```

Comment:
I suppose we could pass the layout instance in in the same way we do the blocks, and ditch using services internally completely?

Reply:
Loking at the code below, we only use it to get the regions? In which case - let's just pass in the regions!

Reply:
Or do we even need to do that as the blocks know what region they're in, do we need empty regions to exist?